src/index.ts

```diff
@@ -0,0 +1,61 @@
+interface Field {
+  left: number;
+  top: number;
+  width: number;
+  height: number;
+  chars: string;
+}
+
+export type Fieldset = Field[];
+
+function getNumberFromWords(words: string[]): string {
+  if(!words.length){
+    throw new Error('Words cannot be an empty array');
+  }
+  const number = words.pop();
+  if(number && number.match(/[A-Za-z\d]/) !== null){
+    return number;
+  }
+  return getNumberFromWords(words);
+}
+
+function parseInvoiceNumber(input: string) {
+  const words = input.split(' ');
+  const number = getNumberFromWords(words);
+  const label = words.join(' ');
+  return {
+    label,
+    number,
+  }
+}
+
+export function main(inputFields: Fieldset) {
+  const newFields: ({index?: number} & Field)[] = [];
+  const outputFields = inputFields.map((field, index) => {
+    if(field.chars.match(/invoice/i) !== null){
```

Comment:
We check here for the correct field. We want to make sure we are in an invoice number field so we check for the regex of `invoice` and make it case-insensitive.